client/src/components/sidebar/ProfilePictureModal.tsx

```diff
@@ -0,0 +1,298 @@
+import React, { useState, useRef, useEffect } from 'react';
+import ReactCrop, { centerCrop, makeAspectCrop, Crop, PixelCrop } from 'react-image-crop';
+import { API_URL } from '../../api/config';
+import CloudUploadIcon from '@mui/icons-material/CloudUpload';
+import Snackbar from '@mui/material/Snackbar';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import { canvasPreview } from './canvasPreview';
+import { useDebounceEffect } from './useDebounceEffect';
+
+import 'react-image-crop/dist/ReactCrop.css';
+import { Button, styled } from '@mui/material';
+
+import DialogTitle from '@mui/material/DialogTitle';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import { useAuth } from '../../hooks/useAuth';
+import { emptyProfile } from './friends/UserProfile';
+
+const MAX_FILE_SIZE_BYTES = 2000000;
+
+const VisuallyHiddenInput = styled('input')({
+  clip: 'rect(0 0 0 0)',
+  clipPath: 'inset(50%)',
+  height: 1,
+  overflow: 'hidden',
+  position: 'absolute',
+  bottom: 0,
+  left: 0,
+  whiteSpace: 'nowrap',
+  width: 1,
+});
+
+const StyledContainer = styled('div')({
+  margin: '20px 0',
+  padding: '0 20px',
+});
+
+const StyledPromptText = styled('div')({
+  fontWeight: 'bold',
+  marginBottom: '10px',
+});
+
+const fileButtonProps = {
+  marginTop: '30px',
+  textTransform: 'none',
+  fontSize: '16px',
+};
+
+const removeButtonProps = {
+  marginTop: '30px',
+  marginLeft: '10px',
+  textTransform: 'none',
+  fontSize: '16px',
+};
+
+const saveButtonProps = {
+  marginTop: '10px',
+  textTransform: 'none',
+  fontSize: '16px',
+  width: '100%',
+};
+
+const StyledReactCrop = styled(ReactCrop)`
+  width: 100%;
+
+  & * {
+    width: 100%;
+  }
+`;
+
+function centerAspectCrop(mediaWidth: number, mediaHeight: number, aspect: number) {
+  return centerCrop(
+    makeAspectCrop(
+      {
+        unit: '%',
+        width: 90,
+      },
+      aspect,
+      mediaWidth,
+      mediaHeight,
+    ),
+    mediaWidth,
+    mediaHeight,
+  );
+}
+
+export default function App() {
```

Comment:
suggestion: functional component naming conventions
```suggestion
export default function ProfilePictureModal() {
```